draft-ietf-dnssd-privreq.xml

```diff
@@ -532,223 +543,171 @@ analysis can often reveal the service.
+    (particularly problematic on battery devices)
+    and may result in the discovery process being slow.</t>
+
+    <t>It is a difficult challenge to design a discovery protocol that has the
+    property of obscuring the details of what it is doing from unauthorized
+    observers, while also managing to do that quickly and efficiently.</t>
+
+  </section>
+
+  <section title="Secure Initialization and Trust Models">
+
+    <t>One of the challenges implicit in the preceding discussions is
+    that whenever we discuss "trusted entities" versus "untrusted entities",
+    there needs to be some way that trust is initially established,
+    to convert an "untrusted entity" into a "trusted entity".</t>
+
+    <t>One way to establish trust between two entities is
+    to trust a third party to make that determination for us.
+    For example, the X.509 certificates used by TLS and HTTPS web browsing
+    are based on the model of trusting a third party to tell us who to trust.
+    There are some difficulties in using this model for establishing
+    trust for service discovery uses.
+    If we want to print our tax returns or medical documents on "our"
+    printer, then we need to know which printer on the network we can
+    trust be be "our" printer.
+    All of the printers we discover on the network may be legitimate
+    printers made by legitimate printer manufacturers, but not all of
+    them are "our" printer. A third-party certificate authority cannot
+    tell us which one of the printers is ours.</t>
+
+    <t>Another common way to establish a trust relationship is
+    Trust On First Use (TOFU), as used by ssh.
+    The first usage is a Leap Of Faith, but after that public keys
+    are exchanged and at least we can confirm that subsequent
+    communications are with the same entity.
+    In today's world, where there may be attackers present even at that
+    first use, it would be preferable to be able to establish a trust relationship
+    without requiring an initial Leap Of Faith.</t>
+
+    <t>Techniques now exist for securely establishing a trust relationship
+    without requiring an initial Leap Of Faith.
+    Trust can be established securely using a short passphrase or PIN with
+    cryptographic algorithms such as
+    <xref target="RFC5054">Secure Remote Password (SRP)</xref>
+    or a Password Authenticated Key Exchange like
+    <xref target="RFC8236">J-PAKE</xref>
+    using a
+    <xref target="RFC8235">Schnorr Non-interactive Zero-Knowledge Proof</xref>.
+    </t>
+
+    <t>Such techniques require a user to enter the correct passphrase or PIN
+    in order for the cryptographic algorithms to establish working communication.
+    This avoids the human tendency to simply press the "OK" button when asked
+    if they want to do something on their electronic device.
+    It removes the human fallibility element from the equation,
+    and avoids the human users inadvertently sabotaging their own security.</t>
+
```

Comment:
Arguably, the "press OK" problem is really an interface design issue. Daniel would say that an SAS exchange could end by requiring the user to copy the PIN from one device to the other one.

I am a bit concerned that the tone here is advocacy for a specific solution, rather than actually expressing the requirements such as proximity or not, leap of faith or not, trust the user to say OK or not, etc. Besides, some of the solutions listed in the scaling draft do not require such procedures. For example, copying the server's public key would just require scanning a QR code, no PIN involved.

Reply:
Hm, yes, I would agree. I can change that tone in another PR, if you like?